weather.go

```diff
@@ -0,0 +1,111 @@
+package weather
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type apiResponse struct {
+	Name    string
+	Weather []struct {
+		Main string
+	}
+	Main struct {
+		Temp float64
+	}
+}
+
+type Conditions struct {
+	City               string
+	OneWord            string
+	TemperatureCelsius float64
+}
+
+func RunCLI(args []string) {
+
+	token := os.Getenv("WEATHERAPP_TOKEN")
+	if token == "" {
+		fmt.Fprintf(os.Stderr, "please set env variable, WEATHERAPP_TOKEN \n")
+		os.Exit(2)
+	}
+
+	request, err := Request(args, token)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "problem setting url', %v", err)
+		os.Exit(2)
+	}
+
+	response, err := Response(request)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "an error has occured, %v", err)
+		os.Exit(2)
+	}
+
```

Comment:
We need to defer closing `response` here, don't we? In fact, it might be better to have `Response` read all the response body (using `io.ReadAll`) into a `[]byte`, and return that instead. It's good to avoid passing any leakable resource (such as a response body) over an API boundary—because it's not always clear whose job it is to close it.